docs/src/charm/tutorial/getting-started.md

```diff
@@ -1 +1,243 @@
+worker node in a similar way to the original K8s node:
+
+```
+juju deploy k8s-worker --channel=latest/edge --constraints='cores=2 mem=16G root-disk=40G'
+```
+
+Once again, this will take a few minutes. In this case though, the `k8s-worker`
+application won't settle into a 'Ready' status, because it requires a
+connection to the control plane. This is handled in Juju by integrating the
+charms so they can communicate using a standard interface. The charm info we
+fetched earlier also includes a list of the relations possible, and from this
+we can see that the k8s-worker requires "cluster: k8s-cluster".
+
+To connect these charms and effectively add the worker to our cluster, we use
+the 'integrate' command, adding the interface we wish to connect
+
+```
+juju integrate k8s k8s-worker:cluster
+```
+
+After a short time, the worker node will share information with the control plane and be joined to the cluster.
+
+## 4. Scale the cluster (Optional)
+
+If one worker doesn't seem like enough, we can easily add more:
+
+```
+juju add-unit k8s-worker -n 1
+```
+
+This will create an additional instance running the k8s-worker charm. Juju
+manages all instances of the same application together, so there is no need to
+add the integration again. If you check the Juju status, you should see that a
+new unit is created, and now you have `k8s-worker/0` and `k8s-worker/1`
+
+
+## 5. Use Kubectl
+
+[Kubectl][] is the standard upstream tool for interacting with a Kubernetes
+cluster. This is the command that can be used to inspect and manage your
+cluster.
+
+If you don't already have `kubectl`, it can be installed from a snap:
+
+```
+sudo snap install kubectl --classic
+```
+
+If you have just installed it, you should also create a file to contain the configuration:
+
+```
+mkdir ~/.kube
+```
+
+To fetch the configuration information from the cluster we can run:
+
+```
+juju run k8s/0 get-kubeconfig 
+```
+
+The Juju action is a piece of code which runs on a unit to perform a specific
+task. In this case it collects the cluster information - the YAML formatted
+details of the cluster and the keys required to connect to it.
+
+```{warning}  If you already have Kubectl and are using it to manage other clusters,
+   you should edit the relevant parts of the cluster yaml output and append them to
+   your current kubeconfig file.
+```
+
+We can use pipe to append your cluster's kubeconfig information directly to a
+config file which will just require a bit of editing:
+
+```
+juju run k8s/0 get-kubeconfig >> ~/.kube/config
+```
+
+The output includes the root of the YAML, `kubeconfig: |`, so we can just use an editor to remove that line:
+
+```
+nano ~/.kube/config
+```
+
+Please use the editor of your choice to delete the first line and save the file.
+
+Alternatively, if you are a `yq` user, the same can be achieved with:
+
+```
+juju run k8s/0 get-kubeconfig | yq '.kubeconfig' -r >> ~/.kube/config
```

Comment:
any concern about the `>>` append to an existing `~/.kube/config`?

Reply:
No, because I would rather it got corrupted  but salvageable than overwrite an existing valid config

Reply:
I feel we have given people enough pointers. If we want to avoid having to explain in detail how to manually construct a kubeconfig then we should do some work on managing it properly.

Reply:
Agree.. Nicely Done.  Thank you.